lib/Dialect/Arc/Transforms/ArcCanonicalizer.cpp

```diff
@@ -657,6 +663,46 @@ MergeVectorizeOps::matchAndRewrite(VectorizeOp vecOp,
   return success();
 }
 
+LogicalResult KeepOneVecOp::matchAndRewrite(VectorizeOp vecOp,
+                                            PatternRewriter &rewriter) const {
+  BitVector argsToRemove(vecOp.getInputs().size(), false);
+  DenseMap<VectorizeOp, SmallVector<unsigned>> opExist;
+  for (auto [argIdx, inputVec] : llvm::enumerate(vecOp.getInputs())) {
+    auto otherVecOp = inputVec[0].getDefiningOp<VectorizeOp>();
```

Comment:
For this canonicalization pattern, it doesn't matter where the values come from, so we can simplify (and at the same time generalize) this a bit.